core/templates/dev/head/pages/exploration-editor-page/feedback-tab/services/thread-data.service.ts

```diff
@@ -51,17 +51,27 @@ angular.module('oppia').factory('ThreadDataService', [
       suggestionThreads: []
     };
 
+    // TODO(brianrodri@): Use a helper-object to give this function O(1)
+    // complexity instead of O(N).
+    var getThreadWithId = function(threadId) {
+      var thread = null;
+      var allThreads = _data.feedbackThreads.concat(_data.suggestionThreads);
+      for (var i = 0; i < allThreads.length; i++) {
+        if (allThreads[i].threadId === threadId) {
+          thread = allThreads[i];
+          break;
+        }
+      }
+      return thread;
+    };
+
     // Number of open threads that need action
     var _openThreadsCount = 0;
 
-    var _fetchThreads = function(successCallback = function() {}) {
+    var _fetchThreads = function(onSuccess) {
```

Comment:
Should we call this _fetchThreadsAsync to signal asynchronicity? Ditto elsewhere for other async methods.